feat: SuperchainWETH redesign

agusduha opened the pull request:
Closes OPT-264

GrapeBaBa committed:
MT Cannon: add mips movf/movt tests (#12392)

* feat:add movf/movt tests

Signed-off-by: Chen Kai <281165273grape@gmail.com>

* fix:fix code review nit

Signed-off-by: Chen Kai <281165273grape@gmail.com>

---------

Signed-off-by: Chen Kai <281165273grape@gmail.com>

maurelian committed:
deployOpChain using OPCM (#12291)

* feat: move log to top of save function

Makes debugging slightly easier.

* feat: Deploy system with OPCM

* feat: Deploy missing Permissionless DWETH

* feat: Fully OPCM based deployment

At this point tests that depend on the L2OutputOracle will still fail,
but those with Fault Proofs _should_ pass.

* feat: Refactor _setFaultGameImplementation

It no longer handles the PERMISSIONED game type deploy by OPCM.

* test: Make caller explicit in unauthorized test case

* test: Fix have guardian setRespectedGameType to Cannon

THis is necessary because the tests assume the respecte game is cannon,
but the OPCM assumes it is the permissioned game

* test: Bridge_Initializer defaulst to FP on

* test: setRespectedGameType to Cannon in Deploy

* test: Bridge_Initializer defaulst to FP on

* feat: resetInitializedProxy func to allow reinitializing contracts

* feat: Delete initializeOpChain, we don't need it anymore

* feat: Set batch inbox address in config using hashed method

* feat: Make caller explicit in testFuzz_removeDependency_notDependencyManager_reverts

* feat: lint

* Revert "test: Bridge_Initializer defaulst to FP on"

This reverts commit d435653b5405f35970e9663cfddd5a7214fcdb57.

* feat: Consolidate useFaultProofs modifications into _run

* Revert "test: Bridge_Initializer defaulst to FP on"

This reverts commit af8d99b94393c3adef32b6ae5d5384e6766569e5.

* feat: Some annotations in Deploy.s.ol

* feat: Skip checking L2OO in Initializable.t.sol

* feat: Delete unused initialize functions

* fix: Remove unused imports

* fix: unused import

* feat: Use respectedGameType from deploy-config

* use startPrank

* feat: Prevent deploying legacy portal for interop

* fix: incorrect var name

* feat: detect when OPCM is deploying the Permissionless game

* fix: error format

jsvisa committed:
feat(contracts-bedrock): remove old forge-std code (#12378)

* feat(contracts-bedrock): rm assume on precompile

Signed-off-by: jsvisa <delweng@gmail.com>

* feat(contracts-bedrock): replace vm.keyExists with vm.keyExistsJson

Signed-off-by: jsvisa <delweng@gmail.com>

* feat(contracts-bedrock): replace _readOr implement

Signed-off-by: jsvisa <delweng@gmail.com>

---------

Signed-off-by: jsvisa <delweng@gmail.com>

mds1 committed:
doc: add more security review info (#12429)

blmalone committed:
op-deployer: Fee Recipients and Gas Params added to intent (#12404)

* op-deployer: Fee Recipients and Gas Params added to intent

* fix: retrieved fee recipients from docs.

* fix: added a loose test for checking fee vault recipients and gas params.

* fix: programmatically retrieving the proxy impl address to check the immutable.

* fix: fee recipient per chain

smartcontracts committed:
feat(ci): add semgrep to contracts checks (#12395)

Adds the semgrep step to contracts-bedrock checks now that semgrep
is added to the latest version of ci-builder.

Commit:
[batcher] derive.ChannelOut factory (#12344)

* Add support for a derive.ChannelOut factory

* Add DriverSetupOption for injecting custom options into the DriverSetup

* Remove factory from NewChannelManager and NewChannelBuilder

* Add ChannelOut factory test

* Add comment about why we use a wrapper

joohhnnn committed:
MTCannon: improve consistency & add EmptyThreadStack test (#12389)

* Add EmptyThreadStacks test

* add go evm check

* improve consistency and update test

* delete emptyThreadedProofGenerator

* forge lint update

* fix solidity versioning and hash & add proof variations in test

trianglesphere committed:
op-service: Add optional headers to the signer client (#12407)

* op-service: Add optional headers to the signer client

* Explain flag usage

Commit:
fix: L1 blobs fetcher interface implementation assertion (#12403)

mbaxter committed:
cannon: Add more load / store tests (#12432)

* cannon: Add a few more load/store test cases

* cannon: Add some more load test cases with negative numbers

* cannon: Specify test names more precisely

agusduha committed:
feat: superweth redesign

ajsutton committed:
op-challenger: Set op-program log level based on the challenger level (#12379)

* op-challenger: Set op-program log level based on the challenger level.

* op-challenger: Fix tests to actually assert values.

dependabot[bot] committed:
dependabot(gomod): bump github.com/urfave/cli/v2 from 2.27.4 to 2.27.5 (#12447)

Bumps [github.com/urfave/cli/v2](https://github.com/urfave/cli) from 2.27.4 to 2.27.5.
- [Release notes](https://github.com/urfave/cli/releases)
- [Changelog](https://github.com/urfave/cli/blob/main/docs/CHANGELOG.md)
- [Commits](https://github.com/urfave/cli/compare/v2.27.4...v2.27.5)

---
updated-dependencies:
- dependency-name: github.com/urfave/cli/v2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Commit:
[batcher] Cleanup batcher channel inclusion block logic (#12363)

* Cleanup batcher channel inclusion block logic

* Add comment to isTimedOut

* Remove updateInclusionBlocks altogether

* Revert receiver variable rename

* Fix tests

* Fix isTimedOut for ChannelTimeouts of 1 (ensure that some txs have been confirmed)

* Added comment about confirmed txs to isFullySubmitted

ajsutton committed:
op-challenger: Support running multiple prestates the same game type in run-trace (#12443)

dependabot[bot] committed:
dependabot(gomod): bump github.com/minio/minio-go/v7 (#12446)

Bumps [github.com/minio/minio-go/v7](https://github.com/minio/minio-go) from 7.0.77 to 7.0.78.
- [Release notes](https://github.com/minio/minio-go/releases)
- [Commits](https://github.com/minio/minio-go/compare/v7.0.77...v7.0.78)

---
updated-dependencies:
- dependency-name: github.com/minio/minio-go/v7
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

0xDiscotech commented on packages/contracts-bedrock/src/L2/SuperchainWETH.sol:
```diff
@@ -37,63 +43,48 @@ contract SuperchainWETH is WETH98, ISuperchainWETH, ISemver {
-
-        // Mint to user's balance.
-        _mint(dst, wad);
+        _mint(_to, _amount);
```
This should go before the call to follow CEI, right?

0xDiscotech replied:
I know this is copy-pasted logic, but to me would be better to follow the pattern if it doesn't harm

0xDiscotech commented on packages/contracts-bedrock/src/L2/SuperchainWETH.sol:
```diff
@@ -9,18 +9,24 @@ import { Predeploys } from "src/libraries/Predeploys.sol";
+
     /// @notice Semantic version.
     /// @custom:semver 1.0.0-beta.6
     string public constant version = "1.0.0-beta.6";
```
Missing change over version

0xDiscotech commented on packages/contracts-bedrock/test/L2/SuperchainWETH.t.sol:
```diff
@@ -121,237 +130,189 @@ contract SuperchainWETH_Test is CommonTest {
-        // Assume
-        vm.assume(_chainId != block.chainid);
+    /// @notice Tests the `crosschainMint` with custom gas token succeeds and emits the `CrosschainMinted` event.
+    function testFuzz_crosschainMint_fromBridgeCustomGasTokenChain_succeeds(address _to, uint256 _amount) public {
```
I think it'd be a good idea to use Foundry's `function expectCall(address callee, bytes calldata data, uint64 count) external;` vm cheatcode setting `count` to zero `mint` call over ETHLiquidity - so we ensure that's not called.

The same could be applied for `crosschainBurn()`

0xDiscotech reviewed (commented):
Looking good ser! 
Left few small comments in addittion that on the tests, the assertions after a revert could be removed, IMO no need to add them after `expectRevert` was used - not a big deal tho

ajsutton committed:
proofs-tools: Update challenger to include new vm runner options. (#12452)

mslipper committed:
ci: Port some CI jobs to self-hosted runners (#12199)

* ci: Port some CI jobs to self-hosted runners

This PR ports the following CI jobs to use self-hosted runners:

- cannon-go-lint-and-test
- contracts-bedrock-build
- go-lint
- go-test-kurtosis
- go-e2e-test
- cannon-prestate

These jobs benefit from running on beefier hardware. Since Go cache data is shared among the executors, it also allowed me to remove a lot of the manual caching logic from our builds while  improving Go compilation times overall.

I'll provide a separate writeup detailing exactly how the executors work.

* use test result directories in the workspace

Commit:
Rename channel receiver (#12453)

agusduha committed:
fix: pr fixes

sebastianst committed:
op-node/rollup/derive: Add Holocene Channel Stage (#12334)

This only adds the new stage, but doesn't wire it into the derivation
pipeline yet.

mslipper committed:
op-e2e: Parallelize CGT tests (#12464)

* op-e2e: Parallelize CGT tests

Parallelizes the custom gas token tests. This reduces runtime from ~120s to ~60s on my Macbook.

* code review updates

maurelian committed:
test: Remove exceptions for unlabelled fdg contracts (#12462)

* test: Remove exceptions for unlabelled fdg contracts

* fix: Unused import

agusduha committed:
feat: introduce SuperchainERC20 redesign + ICrosschainERC20 (#12321)

* feat: add superchain erc20 bridge (#61)

* feat: add superchain erc20 bridge

* fix: interfaces and versions

* refactor: optimism superchain erc20 redesign (#62)

* refactor: use oz upgradeable erc20 as dependency

* chore: update interfaces

* fix: tests based on changes

* refactor: remove op as dependency

* feat: add check for supererc20 bridge on modifier

* chore: update tests and interfaces

* chore: update stack vars name on test

* chore: remove empty gitmodules file

* chore: update superchain weth errors

* test: add superchain erc20 bridge tests (#65)

* test: add superchain erc20 bridge tests

* test: add optimism superchain erc20 beacon tests

* test: remove unnecessary test

* test: tests fixes

* test: tests fixes

* chore: update missing bridge on natspec (#69)

* chore: update missing bridge on natspec

* fix: natspecs

---------

Co-authored-by: agusduha <agusnduha@gmail.com>

* fix: remove superchain erc20 base (#70)

* refactor: update isuperchainweth (#71)


---------

Co-authored-by: agusduha <agusnduha@gmail.com>

* feat: rename mint/burn and add SuperchainERC20 (#74)

* refactor: rename mint and burn functions on superchain erc20

* chore: rename optimism superchain erc20 to superchain erc20

* feat: create optimism superchain erc20 contract

* chore: update natspec and errors

* fix: superchain erc20 tests

* refactor: make superchain erc20 abstract

* refactor: move storage and erc20 metadata functions to implementation

* chore: update interfaces

* chore: update superchain erc20 events

* fix: tests

* fix: natspecs

* fix: add semmver lock and snapshots

* fix: remove unused imports

* fix: natspecs

---------

Co-authored-by: 0xDiscotech <131301107+0xDiscotech@users.noreply.github.com>

* fix: refactor zero check (#76)

* fix: pre pr

* fix: semver natspec check failure (#79)

* fix: semver natspec check failure

* fix: ignore mock contracts in semver natspec script

* fix: error message

* feat: add crosschain erc20 interface (#80)

* feat: add crosschain erc20 interface

* fix: refactor interfaces

* fix: superchain bridge natspec (#83)

* fix: superchain weth natspec (#84)

Co-authored-by: 0xng <ng@defi.sucks>
Co-authored-by: 0xParticle <particle@defi.sucks>
Co-authored-by: gotzenx <78360669+gotzenx@users.noreply.github.com>

* fix: stop inheriting superchain interfaces (#85)

* fix: stop inheriting superchain interfaces

* fix: move events and erros into the implementation

* fix: make superchainERC20 inherits from crosschainERC20

* fix: superchain bridge rename (#86)

* fix: fee vault compiler error (#87)

* fix: remove unused imports

* fix: refactor common errors (#90)

* fix: refactor common errors

* fix: remove unused version

* fix: reuse unauthorized error (#92)

* fix: superchain erc20 factory conflicts

* fix: rename crosschain functions (#94)

---------

Co-authored-by: Disco <131301107+0xDiscotech@users.noreply.github.com>
Co-authored-by: 0xng <ng@defi.sucks>
Co-authored-by: 0xParticle <particle@defi.sucks>
Co-authored-by: gotzenx <78360669+gotzenx@users.noreply.github.com>

mslipper committed:
op-e2e: Suffix Kurtosis enclaves with current time (#12473)

This prevents spurious failures when a test fails to properly clean up enclaves after shutdown.

smartcontracts committed:
maint(ct): fix AddressManager deploy (#12471)

Updates Deploy.s.sol to use create1AndSave for the AddressManager
contract.

protolambda committed:
op-supervisor: cleanup, refactor, local-safe info from op-node (#12427)

* op-supervisor: cleanup, refactor to take local-safe info from op-node

* Refactor ChainProcessor Worker

* remove unneeded err check

* semgrep

---------

Co-authored-by: axelKingsley <axel.kingsley@gmail.com>

protolambda committed:
op-supervisor: dependency-set config (#12450)

* op-supervisor: cleanup, refactor to take local-safe info from op-node

* op-supervisor: dependency-set config

---------

Co-authored-by: axelKingsley <axel.kingsley@gmail.com>

ajsutton committed:
cannon: Add a script to build multicannon with support for legacy versions. (#12455)

protolambda committed:
interop: fromda DB (#12436)

* op-supervisor: cleanup, refactor to take local-safe info from op-node

* interop: draft fromda DB

* op-supervisor: fromda DB fixes and tests

* fix test warnings

* add extra check ; remove LatestDerived, LatestDerivedFrom

---------

Co-authored-by: axelKingsley <axel.kingsley@gmail.com>

ajsutton committed:
op-challenger: Introduce --prestates-url to specify prestates source for all game types at once (#12440)

* op-challenger: Introduce --prestates-url to specify prestates source for all game types at once.

* op-challenger: Add required config validation for asterisc kona.

Allow specify both prestate and prestates url to allow overriding. Prestate takes priority.

* op-challenger: Introduce vm-flag to make default with overrides pattern easier.

* op-challenger: Add vm type info to override flags.

ajsutton committed:
op-challenger: Support file: URLs to download prestates from. (#12441)

* op-challenger: Introduce --prestates-url to specify prestates source for all game types at once.

* op-challenger: Add required config validation for asterisc kona.

Allow specify both prestate and prestates url to allow overriding. Prestate takes priority.

* op-challenger: Introduce vm-flag to make default with overrides pattern easier.

* op-challenger: Add vm type info to override flags.

* op-challenger: Support file: URLs to download prestates from.

Makes it possible to support multiple prestates without needing a HTTP server.

mslipper committed:
ci: Combine all go tests into one job (#12474)

* ci: Combine all go tests into one job

Combines all Go tests into a single job. This allows the runner to decide how to best parallelize the tests, and simplifies our CCI config.

* remove module param

* remove dep from compat

* remove dep

* unquote

* use archival node

* attach workspace

maurelian committed:
Add repo links for sherlock contests (#12470)

maurelian committed:
fix: Allow reading the initialized slot of a Proxy (#12465)

* fix: Allow reading the initialized slot of a Proxy

* fix: unused imports

ajsutton committed:
op-e2e: Increase sequencer window for fault proof tests. (#12479)

Reduces flakiness by giving the batcher more time to produce batches.

sebastianst committed:
op-node/rollup/sequencing: Fix temporary engine error handling (#12258)

Do schedule a next sequencer action even if the building state is empty
in `onEngineTemporaryError`.
This can happen if the previous successful payload cleared the building
state and then when `startBuildingBlock` is entered, it hits a temp
error but never modified the cleared building state, so the temp error
handler would never schedule the next action.

Towards #12240 #12041

zhiqiangxu committed:
cannon: use constant instead of magic value (#12386)

* use constant instead of magic value

* address comments

* address comments

* add RegSyscallErrno

0x00101010 committed:
Implement OverrideLeader & LeaderOverriden method for conductor (#12451)

maurelian committed:
feat: Move checkOutput from DeployOPChainOutput to DeployOPChain script (#12469)

* feat: Move checkOutput from DeployOPChainOutput to DeployOPChain script

* fix: Fix compiler warnings

mslipper committed:
op-deployer: Separate L1 and L2 contract artifacts (#12480)

* op-deployer: Separate L1 and L2 contract artifacts

Adds support for specifying separate artifact URLs for L1 and L2. There are two new fields in the intent - `L1ContractsLocator` and `L2ContractsLocator` - which together replace the `ContractArtifactsURL` and `ContractsRelease` fields. Specifying a file or HTTPS URL for either field will automatically enable dev mode for the deployment. Specifying a `tag://` URL will use the standard deployment for L1 and L2. The default have been set to the following:

- L1: `op-contracts/v1.6.0`
- L2: `op-contracts/v1.7.0-beta.1+l2-contracts`

Fixes https://github.com/ethereum-optimism/platforms-team/issues/337.

* fix test

* fix another test

ajsutton committed:
proofs-tools: Update asterisc (#12492)

mslipper committed:
op-deployer: Package for production (#12491)

* op-deployer: Package for production

This PR:

1. Moves op-deployer into a standalone package at the root of the monorepo.
2. Updates CI given the move above.
3. Adds infrastructure for `goreleaser` so that we can provide pre-build binaries to our users.

* fix error in config

* add dep

* move filter

* filters

* goimports

* remove deployer docker builds

* goimports

* remove superfluous package renames

* use correct monorepo dir

* fix bad package renames

* cancel child context

* use different context in cleanup

* log enclave ID

blmalone committed:
op-deployer: Most implementation addresses not set in state.json when standard release tag used (#12434)

* op-deployer: Most implementation addresses not set in state.json when standard release tag used

* Update op-chain-ops/deployer/pipeline/opchain.go

Co-authored-by: Matt Solomon <matt@mattsolomon.dev>

* fix: parallel execution of commands.

* fix: tests added

* fix: vscode revert

* use channel approach

---------

Co-authored-by: Matt Solomon <matt@mattsolomon.dev>
Co-authored-by: Matthew Slipper <me@matthewslipper.com>

mslipper committed:
op-deployer: Marshal tag-based ArtifactsLocator with correct scheme (#12494)

mslipper committed:
ci: Revert kurtosis test back to CCI runner (#12500)

We're seeing issues with concurrent usage of CCI that are causing builds to fail.

qizhou committed:
feat: op-program supports custom chain config (#12310)

* feat: op-program supports custom chain config

* address comments

* address comments

* update mainnet.go/sepolia.go

* Update op-program/chainconfig/chaincfg.go

Co-authored-by: Adrian Sutton <adrian@symphonious.net>

* more error info on config file not found

---------

Co-authored-by: Qi Zhou <qizhou@ethstorage.io>
Co-authored-by: Adrian Sutton <adrian@symphonious.net>

agusduha committed:
Merge branch 'develop' into feat/superweth-redesign

agusduha committed:
fix: fixes post merge

agusduha commented:
Closing, follows in https://github.com/defi-wonderland/optimism/pull/101
